spec/lib/chat_review_queue_spec.rb

```diff
@@ -0,0 +1,304 @@
+      chat_channel.add(second_flagger)
+      guardian_2 = Guardian.new(second_flagger)
+
+      subject.flag_message(message, guardian_2, ReviewableScore.types[:off_topic])
+
+      reviewable = ReviewableChatMessage.find_by(target: message)
+      scores = reviewable.reviewable_scores
+
+      expect(scores.size).to eq(2)
+      expect(scores.map(&:reviewable_score_type)).to contain_exactly(
+        *ReviewableScore.types.slice(:off_topic, :spam).values,
+      )
+    end
+
+    it "publishes a message to the flagger" do
+      messages =
+        MessageBus
+          .track_publish { subject.flag_message(message, guardian, ReviewableScore.types[:spam]) }
+          .map(&:data)
+
+      self_flag_msg = messages.detect { |m| m["type"] == "self_flagged" }
+
+      expect(self_flag_msg["user_flag_status"]).to eq(ReviewableScore.statuses[:pending])
+      expect(self_flag_msg["chat_message_id"]).to eq(message.id)
+    end
+
+    it "publishes a message to tell staff there is a new reviewable" do
+      messages =
+        MessageBus
+          .track_publish { subject.flag_message(message, guardian, ReviewableScore.types[:spam]) }
+          .map(&:data)
+
+      flag_msg = messages.detect { |m| m["type"] == "flag" }
+      new_reviewable = ReviewableChatMessage.find_by(target: message)
+
+      expect(flag_msg["chat_message_id"]).to eq(message.id)
+      expect(flag_msg["reviewable_id"]).to eq(new_reviewable.id)
+    end
+
+    let(:flag_message) { "I just flagged your chat message..." }
+
+    context "when creating a notify_user flag" do
+      it "creates a companion PM" do
+        subject.flag_message(
+          message,
+          guardian,
+          ReviewableScore.types[:notify_user],
+          message: flag_message,
+        )
+
+        pm_topic =
+          Topic.includes(:posts).find_by(user: guardian.user, archetype: Archetype.private_message)
+        pm_post = pm_topic.first_post
+
+        expect(pm_topic.allowed_users).to include(message.user)
+        expect(pm_topic.subtype).to eq(TopicSubtype.notify_user)
+        expect(pm_post.raw).to include(flag_message)
+        expect(pm_topic.title).to eq("Your chat message in \"#{chat_channel.title(message.user)}\"")
+      end
+
+      it "doesn't create a PM if there is no message" do
+        subject.flag_message(message, guardian, ReviewableScore.types[:notify_user])
+
+        pm_topic =
+          Topic.includes(:posts).find_by(user: guardian.user, archetype: Archetype.private_message)
+
+        expect(pm_topic).to be_nil
+      end
+
+      it "allow staff to tag PM as a warning" do
+        admin = Fabricate(:admin)
+        admin_guardian = Guardian.new(admin)
+
+        subject.flag_message(
+          message,
+          admin_guardian,
+          ReviewableScore.types[:notify_user],
+          message: flag_message,
+          is_warning: true,
+        )
+
+        warning = UserWarning.find_by(user: message.user)
+        expect(warning).to be_present
```

Comment:
Should be equivalent?

```suggestion
        expect(UserWarning.exists?(user: message.user)).to eq(true)
```